rskj-core/src/main/java/co/rsk/pcc/bto/BTOUtilsHelper.java

```diff
@@ -0,0 +1,18 @@
+package co.rsk.pcc.bto;
+
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.pcc.NativeContractIllegalArgumentException;
+
+public class BTOUtilsHelper {
+    public NetworkParameters validateAndExtractNetworkFromExtendedPublicKey(String xpub) {
+        // Network determination is done by starting character
+        // Extended public key must start with a "xpub" (mainnet) or "tpub" (testnet)
+        if (xpub.startsWith("xpub")) {
+            return NetworkParameters.fromID(NetworkParameters.ID_MAINNET);
+        } else if (xpub.startsWith("tpub")) {
+            return NetworkParameters.fromID(NetworkParameters.ID_TESTNET);
+        } else {
```

Comment:
We agreed on not doing this, since that would be limiting the potential uses of the precompiled contract. Also, the BTC network in that case would be the BTC network for the Bridge contract, which is an independent contract that should be non-distinguishable from any other that implements a BTC blockchain.